doc/scaling_spire.md

```diff
@@ -1,43 +1,92 @@
+
+For example, in current Istio, all applications on the service mesh are in the same trust domain thus share a common root of trust. There may be more than one service mesh, or applications in the service mesh communicating to external services that need to be authenticated. The use of Federation enables SPIFFE-compatible systems such as multiple Istio service meshes to securely establish trust for secure cross-mesh and off-mesh communications.
+
+## Federation with OIDC-Provider Systems
+
+SPIRE has a feature to programmatically authenticate on behalf of identified workloads to remote systems such as public cloud provider services and secret stores that are OIDC-Federation compatible.  For example, in the case of Amazon Web Services, a SPIRE identified workload can authenticate and communicate with an AWS S3 Bucket, an AWS RDS instance, or AWS CodePipeline.
+
+The SPIRE OIDC Discovery Provider retrieves a WebPKI certificate using the ACME protocol, which it uses to secure an endpoint that serves an OIDC compatible JWKS bundle and a standard OIDC discovery document. The remote OIDC authenticated service needs then to be configured to locate the endpoint and qualify the WebPKI service.   Once this configuration is in place, the remote system’s IAM policies and roles can be set to map to specific SPIFFE IDs.  The workload, in turn, will talk to the OIDC-authenticated system by sending a JWT-SVID.  The target system then fetches a JWKS from the pre-defined URI which is served by the OIDC Discovery Provider.  The target system uses the JWKS file to validate the  JWT-SVID, and if the SPIFFE ID contained within the JWT-SVID is authorized to access the requested resource, it serves the request.  The workload is then able to access the foreign remote service without possessing any credentials provided by it.
+
+For a configuration reference on the OIDC Discovery Provider, see:
+https://github.com/spiffe/spire/tree/master/support/oidc-discovery-provider
+
+For a detailed tutorial on configuring OIDC Federation to Amazon Web Services, refer to: https://spiffe.io/spire/try/oidc-federation-aws/
 
-# Cluster Sizing Considerations
+# Deployment Sizing Considerations
 
-This table provides a reference for sizing SPIRE Servers in SPIRE deployments. These reference numbers are based off a test environment. They are meant as order-of-magnitude guidelines only, and do not represent a performance guarantee for any particular user environment.  Also, the number of workloads and agents shown do not represent the theoretically possible SPIRE deployment scale. Note that datastore performance is not addressed here, and can potentially limit SPIRE performance. There are many factors that will influence SPIRE performance including, but not limited to:
+Factors to consider when sizing a SPIRE deployment for optimum performance include, but are not limited to, the following:
```

Comment:
For each of these factors, consider a sentence that explains how it influences a sizing decision. eg.}

"Frequency of registration changes - modifying a registration entry may cause large numbers of SVIDs in cache to be evicted and regenerated"